src/tests/tables.test.js

```diff
@@ -3,20 +3,92 @@ const glob = require("glob");
 
 describe("duplicate indices", () => {
   it("should contain unique indices", () => {
+    let errors = [];
+    const fileIndices = {};
+
+    forEachFile((filename, entry) => {
+      if (filename in fileIndices === false) {
+        fileIndices[filename] = new Set();
+      }
+
+      if (fileIndices[filename].has(entry.index)) {
+        errors.push(`${filename}: Index '${entry.index}' already exists.`);
+      }
+
+      fileIndices[filename].add(entry.index);
+    });
+
+    expect(errors).toEqual([]);
+  });
+});
+
+describe("api references", () => {
+  it("should not contain broken links", () => {
     let errors = [];
     let files = glob.sync("src/*.json");
 
-    for (const file of files) {
-      const fileText = fs.readFileSync(file, "utf8");
-      const fileJSON = JSON.parse(fileText);
-      const indices = new Set();
-      fileJSON.forEach((entry) => {
-        if (indices.has(entry.index)) {
-          errors.push(`${file}: Index '${entry.index}' already exists.`);
+    let resources = {};
+
+    forEachFile((filename, entry) => {
+      if (entry.url === undefined) return;
+
+      if (entry.index === undefined) {
+        errors.push(`${filename}: Entry with URL '${entry.url}' should have an index.`);
+      }
+
+      resources[entry.url] = { index: entry.index, name: entry.name };
+    });
+
+    forEachFile((filename, topLevelEntry) => {
```

Comment:
Last one I swear, thoughts on combining these?

```suggestion
    forEachFile((filename, topLevelEntry) => {
      if (topLevelEntry.url === undefined) return;

      if (topLevelEntry.index === undefined) {
        errors.push(`${filename}: Entry with URL '${topLevelEntry.url}' should have an index.`);
      }

      resources[topLevelEntry.url] = { index: topLevelEntry.index, name: topLevelEntry.name };
```

Reply:
(It was not the last, so I guess I lied.)

Reply:
No, we have to populate all defined URLs into `resources` before we can begin the validation pass.

Reply:
My bad! I totally missed that.